src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs

```diff
@@ -101,6 +101,43 @@ internal string WaitOnOperation(OperationResult operationResult)
 
             this.ProgressTrackerObject.UpdateProgress("Complete", 100);
 
+            // Handle async operation status
+            var result = JObject.Parse(operationResult.Value);
+
+            JToken statusToken;
+            if (result.TryGetValue("status", out statusToken))
+            {
+                TerminalProvisioningStates resourceProvisioningState;
+                if (Enum.TryParse(value: statusToken.ToString(), ignoreCase: true, result: out resourceProvisioningState))
+                {
+                    if (resourceProvisioningState == TerminalProvisioningStates.Succeeded ||
+                    resourceProvisioningState == TerminalProvisioningStates.Ready)
+                    {
+                        return this.GetResourcesClient()
+                            .GetResource<JObject>(resourceId, apiVersion, CancellationToken.None)
+                            .Result
+                            .ToString();
+                    } else if (resourceProvisioningState == TerminalProvisioningStates.Failed ||
+                        resourceProvisioningState == TerminalProvisioningStates.Canceled)
+                    {
+                        var error = JObject.Parse(operationResult.Value);
+                        JToken errorToken;
+                        if (error.TryGetValue("error", out errorToken))
+                        {
+                            this.FailedResult(
+                                operationResult,
+                                string.Format("The operation failed because resource is in the: '{0}' state. Please check the logs for more details. The error message: {1}", statusToken, errorToken["message"].ToString()));
```

Comment:
You should log the whole errorToken and not just errorToken["message"].
Please see the schema for ExtendedErrorInfo in our code

Reply:
This is the response I see. The parent "message" includes all details, including the inner detailed message. What is missing?

{
	"status": "Failed",
	"error": {
		"code": "ApplianceDeploymentFailed",
		"message": "The operation to create appliance failed. Please check operations of deployment 'testmabugapp' under resource group '/subscriptions/38b598fc-e57a-423f-b2e7-dc0ddb631f1f/resourcegroups/mamanagedapp12'. Detailed error messages: '{\r\n  \"error\": {\r\n    \"code\": \"DnsRecordInUse\",\r\n    \"message\": \"DNS record testsdk22212a341s6.westus.cloudapp.azure.com is already used by another public IP.\",\r\n    \"details\": []\r\n  }\r\n};At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.'"
	}
}

Reply:
For the above, you are probably logging the entire async operation response. Lets sync up f2f for this.